Makefile.common

```diff
@@ -173,17 +195,28 @@ common-tarball: promu
 	@echo ">> building release tarball"
 	$(PROMU) tarball --prefix $(PREFIX) $(BIN_DIR)
 
-.PHONY: common-docker
-common-docker:
-	docker build -t "$(DOCKER_REPO)/$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG)" .
-
-.PHONY: common-docker-publish
-common-docker-publish:
-	docker push "$(DOCKER_REPO)/$(DOCKER_IMAGE_NAME)"
-
-.PHONY: common-docker-tag-latest
-common-docker-tag-latest:
-	docker tag "$(DOCKER_REPO)/$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG)" "$(DOCKER_REPO)/$(DOCKER_IMAGE_NAME):latest"
+.PHONY: common-docker $(BUILD_DOCKER_ARCHS)
+common-docker: $(BUILD_DOCKER_ARCHS)
+$(BUILD_DOCKER_ARCHS): common-docker-%:
+	docker build -t "$(DOCKER_REPO)/$(DOCKER_$*_IMAGE_NAME):$(DOCKER_IMAGE_TAG)" \
+		--build-arg BASE_IMAGE="$(DOCKER_$*_BASE_IMAGE)" \
+		--build-arg BIN_PATH="$(DOCKER_$*_BIN_PATH)" \
+		.
+
+.PHONY: common-docker-publish $(PUBLISH_DOCKER_ARCHS)
+common-docker-publish: $(PUBLISH_DOCKER_ARCHS)
+$(PUBLISH_DOCKER_ARCHS): common-docker-publish-%:
+	docker push "$(DOCKER_REPO)/$(DOCKER_$*_IMAGE_NAME):$(DOCKER_IMAGE_TAG)"
+
+.PHONY: common-docker-tag-latest $(TAG_DOCKER_ARCHS)
+common-docker-tag-latest: $(TAG_DOCKER_ARCHS)
+$(TAG_DOCKER_ARCHS): common-docker-tag-latest-%:
+	docker tag "$(DOCKER_REPO)/$(DOCKER_$*_IMAGE_NAME):$(DOCKER_IMAGE_TAG)" "$(DOCKER_REPO)/$(DOCKER_$*_IMAGE_NAME):latest"
+
+.PHONY: common-docker-manifest
+common-docker-manifest:
```

Comment:
What's the rational for 'docker manifest'? Haven't used it so not sure we need this.

Reply:
The manifest contains references to all arch images and allows the client to just use prometheus/prometheus:tag to get the respective arch image automatically.

Docs: https://docs.docker.com/edge/engine/reference/commandline/manifest/

Reply:
Ah interesting.. It also sounds like Kubernetes uses the same mechanism? https://kubernetes.io/docs/concepts/containers/images/#building-multi-architecture-images-with-manifests But it's not very clear on that.. but I suppose in that case we'd want to have it. Thanks for clarifying!